demos/fastplotlib/actors/sample_generator.py

```diff
@@ -0,0 +1,47 @@
+from improv.actor import Actor, RunManager
+from datetime import date #used for saving
+import numpy as np
+import time
+import logging; logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class Generator(Actor):
+    """
+    Generate data and puts it in the queue for the processor to take
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.data = None
+        self.name = "Generator"
+        self.frame_index = 0
+    
+    def __str__(self):
+        return f"Name: {self.name}, Data: {self.data}"
+
+    def setup(self):
+        logger.info('Completed setup for Generator')
+
+    def stop(self):
+        print("Generator stopping")
+        return 0
+
+    def runStep(self):
+        """
+        Generates 512 x 512 frame and puts it in the queue for the processor
+        """
+
```

Comment:
@clewis7 generate a 2D sine wave with a wavelength that depends on if the number is even or odd

Reply:
instead of doing two different wavelengths it'd be nicer to see scrolling "bars" of a sine wave, you have have `n_xvals` for 1 period, increase the start x_val on each new frame, and just test that the image looks correct based on the current frame number.